tests/ui/repeat-expr/infer-eager.rs

```diff
@@ -0,0 +1,20 @@
+use std::marker::PhantomData;
+
+struct Foo<T>(PhantomData<T>);
+
+impl Clone for Foo<u8> {
+    fn clone(&self) -> Self {
+        Foo(PhantomData)
+    }
+}
+impl Copy for Foo<u8> {}
+
+fn extract<T, const N: usize>(_: [Foo<T>; N]) -> T {
+    loop {}
+}
+
+fn main() {
+    let x = [Foo(PhantomData); 2];
+    //~^ ERROR: type annotations needed
```

Comment:
always deferring the check is a breaking change in cases where the `Copy` impl has inference constraints that are then used in order to perform method lookup (or any other kind of hir typeck check that relies on matching on a type).

want to crater this to see how breaking it is but it would be very easy to just not break this and think that is my preference :3

Reply:
I think my preference now is actually to either always defer and eat the breaking change or do a less-hacky attempt to avoid the breaking change by introducing a new goal kind for repeat expr checks that can be deferred exactly as long as is necessary and no furthur